src/github.com/cloudfoundry-incubator/webhooks/pkg/webhooks/manager/event.go

```diff
@@ -0,0 +1,230 @@
+func ObjectToMapInterface(obj interface{}) (map[string]interface{}, error) {
+	values := make(map[string]interface{})
+	options, err := json.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(options, &values)
+	if err != nil {
+		return nil, err
+	}
+	return values, nil
+}
+
+func getClient(cfg *rest.Config) (client.Client, error) {
+	glog.Infof("Get client for Apiserver")
+	mgr, err := manager.New(cfg, manager.Options{})
+	if err != nil {
+		glog.Errorf("unable to set up overall controller manager %v", err)
+		return nil, err
+	}
+	options := client.Options{
+		Scheme: mgr.GetScheme(),
+		Mapper: mgr.GetRESTMapper(),
+	}
+	apiserver, err := client.New(cfg, options)
+	if err != nil {
+		glog.Errorf("unable create kubernetes client %v", err)
+		return nil, err
+	}
+	return apiserver, err
+}
+
+func meteringToUnstructured(m *Metering) (*unstructured.Unstructured, error) {
+	values, err := ObjectToMapInterface(m)
+	if err != nil {
+		glog.Errorf("unable convert to map interface %v", err)
+		return nil, err
+	}
+	meteringDoc := &unstructured.Unstructured{}
+	meteringDoc.SetUnstructuredContent(values)
+	meteringDoc.SetKind("Sfevent")
+	meteringDoc.SetAPIVersion("instance.servicefabrik.io/v1alpha1")
+	meteringDoc.SetNamespace("default")
+	meteringDoc.SetName(m.getName())
+	labels := make(map[string]string)
+	labels["meter_state"] = ToBeMetered
+	meteringDoc.SetLabels(labels)
+	return meteringDoc, nil
+}
+
+func (e *Event) getMeteringEvent(opt GenericOptions, signal int) *Metering {
+	return newMetering(opt, e.crd, signal)
+}
+
+func (e *Event) getMeteringEvents() ([]*Metering, error) {
+	options := e.crd.Spec.options
+	lo := e.crd.Status.lastOperation
+	oldAppliedOptions := e.oldCrd.Status.appliedOptions
+	var meteringDocs []*Metering
+
+	if e.isDirector() {
+		glog.Infof("Getting Metering Docs for Director: %s", lo.Type)
+		switch lo.Type {
+		case "update":
+			meteringDocs = append(meteringDocs, e.getMeteringEvent(options, MeterStart))
+			meteringDocs = append(meteringDocs, e.getMeteringEvent(oldAppliedOptions, MeterStop))
+		case "create":
+			meteringDocs = append(meteringDocs, e.getMeteringEvent(options, MeterStart))
+		case "delete":
+			meteringDocs = append(meteringDocs, e.getMeteringEvent(oldAppliedOptions, MeterStop))
+		}
+	} else if e.isDocker() {
+		glog.Infof("Getting Metering Docs for Docker : %s", e.crd.Status.State)
+		switch e.crd.Status.State {
+		case "succeeded":
+			meteringDocs = append(meteringDocs, e.getMeteringEvent(options, MeterStart))
+		case "delete":
+			meteringDocs = append(meteringDocs, e.getMeteringEvent(oldAppliedOptions, MeterStop))
+		}
+	}
+	return meteringDocs, nil
+}
+
+func (e *Event) createMertering(cfg *rest.Config) error {
+	apiserver, err := getClient(cfg)
+	if err != nil {
+		return err
+	}
+	events, err := e.getMeteringEvents()
+	if err != nil {
+		return err
+	}
+	for _, evt := range events {
+		unstructuredDoc, err := meteringToUnstructured(evt)
```

Comment:
You can use https://github.com/kubernetes/code-generator to generate a client-go client for Metering and you can use that. It is not recommended to use Unstructured. Of you can use kubebuilder to generate Metering crd and it auto generates the client (client from controller-runtime).

Reply:
Removed the use of unstructured.
Code changes at:
https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/561